packages/atomic/cypress/integration/facets/category-facet.cypress.ts

```diff
@@ -0,0 +1,456 @@
+  assertNonZeroFacetCount,
+  assertClickShowMore,
+  assertShowMoreUA,
+  assertClickShowLess,
+  assertShowLessUA,
+} from './facet-utils';
+import {
+  convertArrayToString,
+  doSortAlphanumeric,
+} from '../../utils/componentUtils';
+
+const categoryFacetProp = {
+  field: 'geographicalhierarchy',
+  label: 'Atlas',
+  defaultCategoryLabel: 'All Categories',
+  numberOfValue: 5,
+  numberOfFacetValueAfterExpand: 7,
+};
+
+const canadaHierarchy = ['North America', 'Canada', 'Alberta'];
+function setupCategoryFacet(field: string, label: string, option?: {}) {
+  setUpPage(`
+    <atomic-breadcrumb-manager></atomic-breadcrumb-manager>    
+    <atomic-category-facet field="${field}" label="${label}" ${option}></atomic-category-facet>`);
+}
+
+function assertClearAllTitleAndTotalParents(
+  clearAllTitle: string,
+  totalParents: number
+) {
+  cy.get(FacetAlias.facetShadow)
+    .find(FacetSelectors.categoryFacetClearLevelButton)
+    .should('contain.text', clearAllTitle);
+  cy.get(FacetAlias.facetShadow)
+    .find('ul[part="parents"] li')
+    .its('length')
+    .should('eq', totalParents);
+}
+
+function assertTotalChildrentMoreThan(value: number) {
+  cy.get(FacetAlias.facetShadow)
+    .find('ul')
+    .not('[part="parents"]')
+    .find('li')
+    .its('length')
+    .should('be.gt', value);
+}
+
+function clickOnCategoryFacetWithValue(value: string) {
+  cy.get(FacetAlias.facetUL)
+    .find(FacetSelectors.categoryFacetNextLevelButton)
+    .contains(value)
+    .click();
+}
+
+describe('Category Facet with default setting', () => {
+  beforeEach(() => {
+    setupCategoryFacet(categoryFacetProp.field, categoryFacetProp.label);
+    createAliasShadow(categoryFacetProp.field, FacetSelectors.categoryFacet);
+    createAliasFacetUL(categoryFacetProp.field, FacetSelectors.categoryFacet);
+    cy.wait('@coveoAnalytics');
+  });
+
+  it('Should load, pass accessibility test, have correct label, facetCount and no facetSearch', () => {
+    validateFacetComponentLoaded(
+      categoryFacetProp.label,
+      FacetSelectors.categoryFacet
+    );
+    cy.get(FacetAlias.facetShadow)
+      .find(FacetSelectors.facetSearchbox)
+      .should('not.exist');
+  });
+
+  it('Should contain ShowMore buttons but should not contain "All Categories" button', () => {
+    cy.get(FacetAlias.facetShadow)
+      .find(FacetSelectors.categoryFacetClearLevelButton)
+      .should('not.exist');
+    cy.get(FacetAlias.facetShadow)
+      .find(FacetSelectors.showMoreButton)
+      .should('be.visible');
+  });
+
+  it('Should not display facetValue in Alphanumeric order', () => {
+    cy.getTextOfAllElements(FacetAlias.facetAllValueLabel).then(
+      (originalValues) => {
+        expect(originalValues).not.to.eql(doSortAlphanumeric(originalValues));
+      }
+    );
+  });
+
+  describe('When user clicks on Arrow to go to next level', () => {
+    beforeEach(() => {
+      clickOnCategoryFacetWithValue(canadaHierarchy[0]);
+      cy.wait(500);
```

Comment:
I will change to `cy.wait('@coveoSearch')`

Reply:
for `beforeEach`, as each 'it...', it starts the test from fresh, so we kind of lost the page injection/precondition, therefore test will fail. Unless we want to include all the checks into 1 call, like the way that I included in the text below, include all checks at the beginning of the test `'Should load, pass accessibility test, have correct label, facetCount and have facet searchbox'`

Reply:
We could add the checks inside one call, but this could make it harder to understand why the test failed.
Another option could be to only put tests that do not change the precondition inside the `describe` block using the `beforeAll`, and then using separate `it` blocks for the tests that change the precondition.